File: recipes-service/src/main/java/com/recipes/recipes_service/domain/Ingredient.java
package com.recipes.recipes_service.domain;

import com.recipes.recipes_service.enums.Unit;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Ingredient {
    private double quantity;
    private Unit unit;
    private String name;
}

File: recipes-frontend/src/app/models/ingredient.ts
import { Unit } from '../enums/unit.enums';

export class Ingredient {
  quantity: number;
  unit: string;
  name: string;

  constructor(
    name: string = '',
    quantity: number = 0,
    unit: string = Unit.GRAMS
  ) {
    this.quantity = quantity;
    this.unit = unit;
    this.name = name;
  }
}
